File: shiftcrypt.py
def caesar_encrypt(plaintext, shift):
    encrypted_text = ""
    for char in plaintext:
        if char.isalpha():
            # Shift only alphabetical characters
            shifted_char = chr(((ord(char.lower()) - ord('a') + shift) % 26) + ord('a'))
            encrypted_text += shifted_char.upper() if char.isupper() else shifted_char
        else:
            # Keep non-alphabetical characters unchanged
            encrypted_text += char
    return encrypted_text

def caesar_decrypt(ciphertext, shift):
    return caesar_encrypt(ciphertext, -shift)

File: shiftcrypt.js
function caesarEncrypt(plaintext, shift) {
    let encryptedText = "";
    for (let char of plaintext) {
        if (/[a-zA-Z]/.test(char)) {
            // Shift only alphabetical characters
            let base = char === char.toLowerCase() ? 'a'.charCodeAt(0) : 'A'.charCodeAt(0);
            let shiftedChar = String.fromCharCode((char.charCodeAt(0) - base + shift) % 26 + base);
            encryptedText += shiftedChar;
        } else {
            // Keep non-alphabetical characters unchanged
            encryptedText += char;
        }
    }
    return encryptedText;
}

function caesarDecrypt(ciphertext, shift) {
    return caesarEncrypt(ciphertext, -shift);
}
